src/http-error-dialog/index.tsx

```diff
@@ -8,14 +8,53 @@ import {
   DialogButton,
   DialogProps,
 } from '@rmwc/dialog';
-import { ApiCallResult } from '@tutorbook/model';
+import { ApiCallError } from '@tutorbook/custom-errors';
 import { Typography } from '@rmwc/typography';
 import { IconButton } from '@rmwc/icon-button';
 
 import styles from './http-error-dialog.module.scss';
 
+const msgs: Record<string, MessageDescriptor> = defineMessages({
+  type: {
+    id: 'http-error-dialog.error-type',
+    defaultMessage: 'A {type} error occurred',
+  },
+  show: {
+    id: 'http-error-dialog.details-show',
+    defaultMessage: 'Show details',
+  },
+  hide: {
+    id: 'http-error-dialog.details-hide',
+    defaultMessage: 'Hide details',
+  },
+  suggestion: {
+    id: 'http-error-dialog.suggestion',
+    defaultMessage: 'You can try later, or contact the support team.',
+  },
+  report: {
+    id: 'http-error-dialog.suggestion-report',
+    defaultMessage: "Press 'Help' to get support.",
+  },
+  code: {
+    id: 'http-error-dialog.error-code',
+    defaultMessage: 'Response status code: {code}',
+  },
+  details: {
+    id: 'http-error-dialog.error-details',
+    defaultMessage: 'Error: {details}',
+  },
+  close: {
+    id: 'btn-tag.close',
+    defaultMessage: 'Close',
+  },
+  help: {
+    id: 'btn-tag.help',
+    defaultMessage: 'Help',
+  },
+});
+
 interface HttpErrorDialogProps extends DialogProps {
-  error: ApiCallResult;
+  error: ApiCallError;
 }
 
 export const HttpErrorDialog: FC<HttpErrorDialogProps> = ({
```

Comment:
Is there a reason why you're not using the `function` keyword? If anything, could you use it for consistency (as I'm using it everywhere else)?